UX_PRINCIPLES.md

```diff
@@ -0,0 +1,78 @@
+### Build Consistency
+- Using the noun before the verb fosters consistency; you also get help at the noun level (imagine your command set as a tree hierarchy branching out from the root); if you can remember the nouns then you can quickly get the list of verbs for that noun (e.g. hab package -help)
+- Use shortcuts sparingly - they’re certainly useful but obstruct understanding and meaning
+- Don’t treat flags as 2nd class citizens. Be aware of existing flags and consider how they can/will be used across multiple commands and subcommands.
+### Make it Readable
+- Familiarity, simplicity and readability make it memorable. Docs are useful, but we don’t want people having to continually refer back to them.
+- Keep commands human readable and memorizable by using a sentence like structure `hab <noun> <verb>`
+- The consistent noun/verb format becomes rhythmic and builds a mental model if used properly
+- Use terms (nouns and verbs) that fit with a real-world analogy
+- Be precise with meaning and pay attention to expectations when choosing words. For example, if you use a command like `hab pkg create` then the expectation should match the result (i.e. an actual package should be created, not simply a directory or scaffold).
+- Don’t make up words. Consider using a different term or add a dash instead of jamming words together or using words that don’t exist in the dictionary. People won’t remember how to spell it which in turn generates unnecessary confusion. Clarity is king.
+- Be careful with abbreviations - keep things speakable - imagine you are telling somebody else to run a command verbally, can they understand/interpret what you say without misspelling it? (this probably means it’s also easier for you to recall on your own as well)
+- Look out for flags that contain boolean wording. For example, try to avoid flags like `thing_disabled` or `thing_enabled`, or even worse, a boolean-ish flag that also takes a boolean value such as hab foo thing_disabled=false or hab foo thing_enabled=true.
+ - Instead, take out boolean wording and just use the setting name like hab foo some_option=enabled or hab foo some_option=false
+### Kill Your Darlings
+- If something feels confusing (even if it’s just single term), then get consensus and change it
+- Don’t ignore and postpone obvious shortcomings. Others see it too, as will users, so take the time to fix it
+- “Fixing it” doesn’t have to change the underlying code. If possible, simply change the part exposed to the user (e.g. it might be called ‘Projects’ in your codebase, but to users it’s all ‘Packages’)
+- Don’t expose your ‘inside baseball’ to users. Re-word things for them and avoid jargon or overly complex technical terminology.
+- As features develop and grow over time, the concepts and terminology may begin to break down and lead to confusion. Don’t be afraid to change a command later if it no longer makes sense.
```

Comment:
It might be worth calling out a change like this in the Habitat documentation. 
ex:

        `hab foo` is deprecated, please use `hab bar`